doc/changelog/08-cli-tools/14238-delete-error.rst

```diff
@@ -0,0 +1,4 @@
+- **Changed:**
+  Makefiles produced by ``coq_makefile`` now use ``.DELETE_ON_ERROR``
```

Comment:
Do you want to mention KEEP_ERROR here or in the reference manual?

Reply:
Oh right I forgot I did that.
TBH I'm not sure it's even worth keeping, WDYT?

Reply:
I think it's reasonable to keep, especially if people want to inspect intermediates or debug things

Reply:
In the manual we say https://coq.github.io/doc/master/refman/practical-tools/utilities.html#coqmakefile-local
>The variable must belong to the variables listed in the Parameters section of the generated makefile. Here we describe only few of them.

KEEP_ERROR isn't important enough to add to the list IMO, so no need for more doc